src/components/Forms/EditProductForm/index.tsx

```diff
@@ -0,0 +1,214 @@
+import { updateProductImage } from '@/utils/api/updateProductImage';
+import { ImageProps } from '@/utils/types';
+import {
+  productSchema,
+  ProductData,
+} from '@/utils/validations/createProductFormValidation';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+
+import ButtonSubmit from '../../ButtonSubmit';
+import DefaultInput from '../../DefaultInput';
+import DefaultSelect from '../../DefaultSelect';
+import UploadImageInput from '../../UploadImageInput';
+import styles from './styles.module.css';
+
+export interface EditProductFormProps {
+  setShowDialog: (value: boolean) => void;
+  product: Product;
+  categoryId: string;
+  editProductFromList: (newProduct: Product, categoryId: string) => void;
+}
+
+export default function EditProductForm({
+  setShowDialog,
+  product,
+  categoryId,
+  editProductFromList,
+}: EditProductFormProps) {
+  const [requestError, setRequestError] = useState(false);
+  const [isSubmiting, setIsSubmiting] = useState(false);
+  const [registredWithSucess, setRegistredWithSucess] = useState(false);
+  const [imagePlaceholder, setImagePlaceholder] = useState<string>(``);
+  const [productImage, setProductImage] = useState<ImageProps>(
+    {} as unknown as ImageProps
+  );
+  const {
+    handleSubmit,
+    register,
+    control,
+    formState: { errors },
+  } = useForm<ProductData>({
+    resolver: yupResolver(productSchema),
+    mode: `onChange`,
+    defaultValues: {
+      name: product.name,
+      description: product.description,
+      price: product.price,
+      categoryId,
+    },
+  });
+
+  const productName = useWatch({
+    control,
+    name: `name`,
+  });
+
+  const onSubmit = async (values: ProductData) => {
+    setIsSubmiting(true);
+
+    const editProductPayload = {
+      ...values,
+      price: Number(values.price),
+    };
+
+    try {
+      const editedProduct = await editProduct({
+        id: product.id,
+        ...editProductPayload,
+      });
+
+      const updateImage = await updateProductImage(
+        product.id,
+        product.productsImages[0].id,
+        [productImage]
+      );
+
+      if (!editedProduct?.id || updateImage.error) {
+        setIsSubmiting(false);
+        return setRequestError(true);
+      }
+
+      const updatedProduct = {
+        ...editedProduct,
+        productsImages: updateImage.productsImages,
+      };
+
+      editProductFromList(updatedProduct, categoryId);
+    } catch (error) {
+      setIsSubmiting(false);
+      setRequestError(true);
+    } finally {
+      setIsSubmiting(false);
+    }
+
+    setRegistredWithSucess(true);
+    setTimeout(() => {
+      setShowDialog(false);
+    }, 2000);
+  };
+
+  const toBase64 = (file: File | undefined) =>
+    new Promise((resolve, reject) => {
+      // vai sair daqui
+      if (file) {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onload = () => resolve(reader.result);
+        reader.onerror = reject;
+      }
+    });
```

Comment:
I think thats Ton already create this function on utils file, and maybe you can use here. The function name is convertFileToBase64, if works you can change too on CreateProductForm